### PR TITLE
Sonar quality enhancement - Batch 1

### PR DESCRIPTION
**Pull request by Chenjp**
Sonar quality enhancement.

## What is the purpose of the change

SonarCloud code quality enhancement.

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-common/src/main/java/org/apache/dubbo/common/beans/factory/ScopeBeanFactory.java [hunk 1]
```diff
@@ -124,19 +110,16 @@ public <T> T getOrRegisterBean(Class<T> type) {
...
-    public <T> T getOrRegisterBean(String name, Class<T> type) {
-        T bean = getBean(name, type);
-        if (bean == null) {
-            // lock by type
-            synchronized (type) {
-                bean = getBean(name, type);
-                if (bean == null) {
-                    bean = createAndRegisterBean(name, type);
-                }
+    private static final Hashtable<Class<?>, ReentrantLock> typeLocks = new Hashtable<>();
```
Comment (AlbumenJ) [inline dubbo-common/src/main/java/org/apache/dubbo/common/beans/factory/ScopeBeanFactory.java]: Why not use `ConcurrentHashMap`?

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-common/src/main/java/org/apache/dubbo/common/beans/factory/ScopeBeanFactory.java (hunk 1 above)
Comment (Chenjp) [reply dubbo-common/src/main/java/org/apache/dubbo/common/beans/factory/ScopeBeanFactory.java]: new commit: replace hashtable with concurrenthashmap.

File: dubbo-common/src/main/java/org/apache/dubbo/common/config/ConfigurationUtils.java [hunk 2]
```diff
@@ -280,9 +280,7 @@ private static <V extends Object> Map<String, V> getSubProperties(
 
         if (CollectionUtils.isNotEmptyMap(configMap)) {
             Map<String, V> copy;
-            synchronized (configMap) {
```
Comment (AlbumenJ) [inline dubbo-common/src/main/java/org/apache/dubbo/common/config/ConfigurationUtils.java:283]: Why remove this sync

File: dubbo-common/src/main/java/org/apache/dubbo/common/threadpool/support/AbortPolicyWithReport.java [hunk 3]
```diff
@@ -145,11 +145,19 @@ public void dispatchThreadPoolExhaustedEvent(String msg) {
...
     }
 
+    public static long getLastPrintTime() {
+        return lastPrintTime;
+    }
+
+    public static void setLastPrintTime(long lastPrintTime) {
+        AbortPolicyWithReport.lastPrintTime = lastPrintTime;
+    }
+
```
Comment (AlbumenJ) [inline dubbo-common/src/main/java/org/apache/dubbo/common/threadpool/support/AbortPolicyWithReport.java:155]: meanless here

File: dubbo-compatible/src/main/java/com/alibaba/dubbo/container/page/PageServlet.java [hunk 4]
```diff
@@ -56,14 +56,18 @@ public static PageServlet getInstance() {
         return INSTANCE;
     }
 
+    private static void setInstance(PageServlet iNSTANCE) {
+        INSTANCE = iNSTANCE;
+    }
+
```
Comment (AlbumenJ) [inline dubbo-compatible/src/main/java/com/alibaba/dubbo/container/page/PageServlet.java:62]: Still meanless

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Comment by oxsean** [comment]
I don't think following these sonar rules makes the code any clearer, like the empty method, the static field is very clean without misunderstandings.

**Comment by Chenjp** [comment]
> I don't think following these sonar rules makes the code any clearer, like the empty method, the static field is very clean without misunderstandings.

How about to disable those meaningless rules in sonar admin?

**Comment by Chenjp** [comment]
This PR is not a bug fix.
Since sonar quality rule is a nice-to-have item. Close the PR.

**Comment by oxsean** [comment]
> > I don't think following these sonar rules makes the code any clearer, like the empty method, the static field is very clean without misunderstandings.
> 
> 
> 
> How about to disable those meaningless rules in sonar admin?

I don't know much about it either. If you have time, could you look into it?

**Pull request closed without merging**